code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -0,0 +1,703 @@
+openapi: 0.1.0
+info:
+  title: Edge Cloud API
+  description:  |
+
+    Edge Cloud API for artifact managment, application management and resource management.  
+    # Introduction
+    APIs defined in this version of the specification can be categorized into the following areas:
+    * __Application Management__ - Deploy and remove applications within artifacts
+    * __Application Artifacts Management__ - Management of application descriptors, binaries, charts and packages 
+    * __MEC Platform information__ - Retrieves all the Edge nodes available according to some defined parameters
+    
+    # Relevant terms and definitions
+    This section provides definitions of terminologies commonly referred to throughout the API descriptions.
+
+    * __Application Provider__ - An application developer, onboarding his/her edge application on operator platform (MEC).    
+    * __Artifact__ - Descriptor, binary image, charts or any other package assosiated with the application.
+    * __OP__ - Operator Platform.
+    * __Region__ - Human readable name of the geografical zone of the MEC. Defined by the OP.
+    
+    # API Functionality
+    __Application Management__
+    * __onboardApplication__ - Submits an application details to an OP. Based on the details provided,  OP shall do bookkeeping, resource validation and other pre-deployment operations. 
+    * __removeApplication__ - Removes an application from an OP
+
+    __Application Artifacts Management__
+    * __onboardArtifact__  Submits an artifact to operator platform
+    * __removeArtifact__  Removes an artifact from operator platform
+    
+    __MEC Platform information__
+    *  __retrieveEdgeNodes__ List of the operator’s MECs and their status, ordering the results by location and filtering by status (active/inactive/unknown)
+       
+    # Further info and support
+  
+    (FAQs will be added in a later version of the documentation)
+
+  termsOfService: http://swagger.io/terms/
+  contact:
+    email: sp-edc@lists.camaraproject.org
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: 1.0.0
+externalDocs:
+  description: Product documentation at Camara
+  url: https://github.com/camaraproject/
+
+servers:
+  - url: "{apiRoot}/{basePath}"
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: API root
+      basePath:
+        default: edge-cloud/v1
+        description: Base path for the Edge Cloud API
+
+security:
+  - oAuth2ClientCredentials: 
+    - nbi-mgmt
+
+tags:
+  - name: Applications
+    description: Manage of Applications
+  - name: Application Artifacts 
+    description: Manage of Artifacts
+  - name: Mobile Edge Cloud Platform
+    description: Mobile Edge Nodes Available
+
+paths:
+  /app:
```

Comment:
The /app endpoints seems to be provisioning an application. But the application itself is not defined. Though it appears that the artefact has been treated as application. So i would propose to make some changes to make the parameters name consistent and if they refer to a single functionality then make their name same. Could be either of artefact or application.

Reply:
That's right! for now it seem to be the same entities and might be a bit confusing. Maybe a further documentation could be provided within the yaml. Anyhow it is planned that in future definitions we would have three separated entities:
-  Application 
- Application instance
- Artifact
Thats why we as Telefonica, believe it might be better to keep it segregated for now

Reply:
We find this issue needed to be discussed in the next meeting as the neccesity to differentiate the Artifact from the application may not be trivial. Our point of view is that an app may need of multiple artifacts to work so maybe a future chang that might be needed in our definition is that the application would accept an array of `artifactIds`. Anyhow in GSMA the artifact and application refers to a unified element (_edgeApplicationManifest_) as you are proposing. Also in the hyperscalers this applies, they usually query a single template of the service to be deployed including the app instance and the artifacts in the same json request.

We can discuss it!

Reply:
As discussed in the previous CAMARA - Edge Cloud meeting (12th - Dec) we agree to combine Artifact and Application into a single type, included in this new version. 

We clarified the difference between Application and Application Instance, defining two methods. Application is HOW to deploy meanwhile Application Instance is WHERE to deploy it.